test/test_container_token_manager.py

```diff
@@ -0,0 +1,275 @@
+            assert request.headers['Authorization'] == 'Basic Y2xpZW50LWlkLTE6Y2xpZW50LXNlY3JldC0x'
+            access_token = TEST_ACCESS_TOKEN_1
+        else:
+            access_token = TEST_ACCESS_TOKEN_1
+
+        response = json.dumps({
+            'access_token': access_token,
+            'token_type': 'Bearer',
+            'expires_in': 3600,
+            'expiration': _get_current_time()+3600,
+            'refresh_token': TEST_REFRESH_TOKEN,
+        })
+
+        return (status_code, {}, response)
+
+    @responses.activate
+    def wrapper():
+        response = responses.CallbackResponse(
+            method=responses.POST,
+            url='https://iam.cloud.ibm.com/identity/token',
+            callback=callback,
+        )
+
+        responses.add(response)
+
+        func()
+
+    return wrapper
+
+
+@mock_iam_response
+def test_request_token_auth_default():
+    iam_url = "https://iam.cloud.ibm.com/identity/token"
+
+    token_manager = ContainerTokenManager(
+        cr_token_filename=cr_token_file,
+        iam_profile_name=MOCK_IAM_PROFILE_NAME,
+    )
+    token_manager.request_token()
+
+    assert len(responses.calls) == 1
+    assert responses.calls[0].request.url == iam_url
+    assert responses.calls[0].request.headers.get('Authorization') is None
+    assert json.loads(responses.calls[0].response.text)['access_token'] == TEST_ACCESS_TOKEN_1
+
+
+@mock_iam_response
+def test_request_token_auth_in_ctor():
+    default_auth_header = 'Basic Yng6Yng='
+    token_manager = ContainerTokenManager(
+        cr_token_filename=cr_token_file,
+        iam_profile_name=MOCK_IAM_PROFILE_NAME,
+        client_id='foo',
+        client_secret='bar')
+
+    token_manager.request_token()
+
+    assert len(responses.calls) == 1
+    assert responses.calls[0].request.headers['Authorization'] != default_auth_header
+    assert json.loads(responses.calls[0].response.text)['access_token'] == TEST_ACCESS_TOKEN_1
+    assert 'scope' not in responses.calls[0].response.request.body
+
+
+@mock_iam_response
+def test_request_token_auth_in_ctor_with_scope():
+    default_auth_header = 'Basic Yng6Yng='
+    token_manager = ContainerTokenManager(
+        cr_token_filename=cr_token_file,
+        iam_profile_name=MOCK_IAM_PROFILE_NAME,
+        client_id='foo',
+        client_secret='bar',
+        scope='john snow')
+
+    token_manager.request_token()
+
+    assert len(responses.calls) == 1
+    assert responses.calls[0].request.headers['Authorization'] != default_auth_header
+    assert json.loads(responses.calls[0].response.text)['access_token'] == TEST_ACCESS_TOKEN_1
+    assert 'scope=john+snow' in responses.calls[0].response.request.body
+
+
+def test_retrieve_cr_token_success():
+    token_manager = ContainerTokenManager(
+        cr_token_filename=cr_token_file,
+    )
+
+    cr_token = token_manager.retrieve_cr_token()
+
+    assert cr_token == 'cr-token-1'
+
+
+def test_retrieve_cr_token_success_fail():
```

Comment:
```suggestion
def test_retrieve_cr_token_fail():
```
?
Are you trying to test one of those "success fail" scenarios? :)

Reply:
Yeah, I am trying to test scenarios that don't even exist. (Just kidding :) )